lib/rust/ensogl/core/src/display/object/class.rs

```diff
@@ -568,7 +568,7 @@ impl<Host> Model<Host> {
     /// previously and is currently attached to .
     pub fn set_on_scene_layer_changed<F>(&self, f: F)
     where F: Fn(&Host, &[WeakLayer], &[WeakLayer]) + 'static {
-        self.callbacks.on_scene_layers_changed.set(Box::new(f))
+        self.callbacks.on_scene_layers_changed.set_if_empty_or_panic(Box::new(f))
```

Comment:
This is my code, but please replace it with `set_if_empty_or_warn` instead and use `WARNING!` instead of logger inside of the function. Also, remove the `or_panic` version. We want the app not to crash if there is even a slight chance of rescuing it.